Cybersource: Update NT flow

almalee24 opened the pull request:
Remote:
135 tests, 670 assertions, 5 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 96.2963% passed

Unit:
142 tests, 782 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

aenand commented on lib/active_merchant/billing/gateways/cyber_source.rb:
```diff
@@ -841,49 +840,18 @@ def subsequent_nt_apple_pay_auth(source, options)
-          cryptogram = Base64.decode64(payment_method.payment_cryptogram)
-          xml.tag! 'ccAuthService', { 'run' => 'true' } do
-            xml.tag!('cavv', Base64.encode64(cryptogram[0...20]))
-            xml.tag!('commerceIndicator', ECI_BRAND_MAPPING[brand])
```
Why remove the commerce_indicator?

almalee24 replied:
Because it's optional and I decided just to remove it.

aenand replied:
Hmmm my concern with removing it is that we needed to mark commerce_indicator as `internet` for recurring AP/GP recently. Do we know what the default commerce_indicator is?

almalee24 replied:
Okay, I'll add it back then.

aenand replied:
Where did you see it was optional? We previously had a brand_mapping for this field so I'm not sure if that mapping was needed or if hard coding `internet` is correct

aenand reviewed (approved):
Thanks for the explanation of commerce indicator! Let's watch this one closely 👀

aenand reviewed (commented):
This looks great!! Is there any remote test(s) that can cover our expected change or do we already have remote tests for NT/AP/GP so it was more ensuring everything passes?

sinourain reviewed (approved):
LGTM, I agree with Amit and Dustin's request, the remote tests we have do not go through the `network_token` [condition](https://github.com/activemerchant/active_merchant/pull/5091/files#diff-ec0894a572ca922f933a687ef23008524682ccea87eec57f3175e7e1c4013bb2R701)

almalee24 commented:
I've updated the NTs payment methods to include `source: :network_token` and I've also added @mastercard_network_token and they're all passing. I didn't add any ApplePay or GooglePay tests because this code doesn't affect those PMs and they logic for that will all stay the same.

aenand reviewed (approved):
Nice :shipit:

sinourain commented:
> I've updated the NTs payment methods to include `source: :network_token` and I've also added @mastercard_network_token and they're all passing. I didn't add any ApplePay or GooglePay tests because this code doesn't affect those PMs and the logic for that will all stay the same.

I ran the tests and was noticing that `payment_method.brand` returns `master` instead of `mastercard` as mapped, this can cause a problem

almalee24 commented:
@aenand @sinourain @DustinHaefele I made a small update to make transactionType 3 for NTs https://github.com/activemerchant/active_merchant/pull/5091/files#diff-ec0894a572ca922f933a687ef23008524682ccea87eec57f3175e7e1c4013bb2L887

sinourain reviewed (approved):
LGTM!

JesseBowling commented on lib/active_merchant/billing/gateways/cyber_source.rb:
```diff
@@ -835,14 +851,26 @@ def network_tokenization?(payment_method)
-        return unless network_tokenization?(payment_method)
+        brand = card_brand(payment_method).to_sym
+        case brand
+        when :visa, :master, :american_express
```
The documentation calls out additional supported brand types [here](https://developer.cybersource.com/docs/cybs/en-us/google-pay/developer/fdiglobal/rest/googlepay/googpay-intro/googpay-processors.html). Is there a reason we're not also adding support for Diners Club, JCB, or Discover here?
